activesupport/test/test_case_test.rb

```diff
@@ -104,7 +104,7 @@ def test_negative_differences
   def test_expression_is_evaluated_in_the_appropriate_scope
     silence_warnings do
       local_scope = "foo"
-      local_scope = local_scope  # to suppress unused variable warning
+      _ = local_scope  # to suppress unused variable warning
```

Comment:
This feels slightly uglier. Does it maybe recognize & special-case this spelling?

```ruby
local_scope = local_scope = "foo"
```

Reply:
`local_scope = local_scope = "foo"` would work for suppessing the ruby warning, but actually the variable is unused and the cop also regard that as unused.

```
% be rubocop activesupport/test/test_case_test.rb
Inspecting 1 file
W

Offenses:

activesupport/test/test_case_test.rb:106:7: W: Lint/UselessAssignment: Useless assignment to variable - local_scope.
      local_scope = local_scope = "foo"
      ^^^^^^^^^^^
activesupport/test/test_case_test.rb:106:21: W: Lint/UselessAssignment: Useless assignment to variable - local_scope.
      local_scope = local_scope = "foo"
                    ^^^^^^^^^^^

1 file inspected, 2 offenses detected
```